### PR TITLE
add @filesystem and @filesystem-ops syscall filter groups

### PR DESCRIPTION
**Pull request by poettering**
After discussions in #4483, this is a new, separate PR now for the `@filesystem` and `@filesystem-ops` syscall filter groups, as they need more discussion and should probably not be merged before v232 is out.

This PR is on top of #4483, hence that PR should really be merged first, and until then please ignore all commits in this PR except the last one.

@keszybz, @topimiettinen, @evverx could you summarize your suggestions regarding these groups here again, please?

### CODE REVIEW

**Comment by keszybz** [comment]
"@resources" looks good.

I don't like the split into "basic" and "advanced" — it's just too arbitrary. (The names "@filesystem" and "@filesystem-ops" also don't really explain the difference). I'd prefer to add new groups "@filesystem-read" and "@filesystem-write", even if if requires looking the arguments of some system calls, not just at the syscall number. @filesystemd-read should include the calls that don't modify the file system, and @filesystem-write the ones that do or may modify. The tricky syscalls would be `open`, `mmap` and such. Ultimately this would be more useful to users than a raw syscall list.

This PR needs rebasing after my `systemd-analyze syscall-filter` additions.

**Comment by poettering** [comment]
OK, I force pushed a new rebased version of this. I merged the two groups into one now: "@filesystem" that is hopefully more acceptable?

I am not sure we could properly distuingish read/write access with seccomp for all cases really. And if we can I am not sure this belongs under SystemCallFilters= (which really just matches system calls so far, but doesn't try to grok semantics of a call), but probably should get its own call ReadOnlyFileAccess=yes|no or so...

("@resources" already got merged btw, as part of the original PR.)

**Comment by keszybz** [comment]
Yeah, let's merge this. If we ever decide to make this more fancy, we can add `@file-system-read` and `@file-system-write` or whatever other options.

The failure is the one we have seen before, probably unrelated:
```
- ['ifup@ens2.service loaded failed failed ifup for ens2']
+ []
```

**Comment by keszybz** [comment]
And also
```
subprocess.CalledProcessError: Command '['/lib/systemd/systemd-networkd-wait-online', '--interface', 'test_eth42', '--timeout=15']' returned non-zero exit status 1
```

**Comment by lucaswerkmeister** [comment]
Is it intentional that the group includes `mmap` but not `munmap`? Seems like a weird asymmetry.
